v-next/hardhat-node-test-reporter/src/error-formatting.ts

```diff
@@ -96,10 +97,27 @@ function formatSingleError(
 
   const diff = getErrorDiff(error);
 
-  const stackReferences: StackReference[] = stackLines
+  let stackReferences: StackReference[] = stackLines
     .map(parseStackLine)
     .filter((reference) => reference !== undefined);
 
+  // Remove all the stack references exposing test runner internals
+  // User code starts after: (Suite|Test|TestHook|...).runInAsyncScope
+  const runInAsyncScopeIndex = stackReferences.findIndex((reference) => {
+    return (
+      reference.context !== undefined &&
+      reference.context.endsWith(".runInAsyncScope")
```

Comment:
Can we make this check a bit stricter?  Maybe also check that the location is `node:...`?

Reply:
Actually, this is just a special case of a `node:...` reference now that I think about it. I think we can do without it altogether.

I'm going to remove it.

Reply:
Oh, that's true! I hadn't thought about it